src/Datadog.Trace/DuckTyping/DuckType.cs

```diff
@@ -151,7 +151,7 @@ private static CreateTypeResult CreateProxyType(Type proxyDefinitionType, Type t
                     }
 
                     // Create a valid type name that can be used as a member of a class. (BenchmarkDotNet fails if is an invalid name)
-                    string proxyTypeName = $"{assembly}.{targetType.FullName.Replace(".", "_").Replace("+", "__")}.{proxyDefinitionType.FullName.Replace(".", "_").Replace("+", "__")}";
+                    string proxyTypeName = $"{assembly}.{targetType.FullName.Replace(".", "_").Replace("+", "__")}.{proxyDefinitionType.FullName.Replace(".", "_").Replace("+", "__")}_{++_typeCount}";
```

Comment:
Don't require `Interlocked` because the CreateProxyType method is called only once from a Concurrent Lazy pattern.